src/features/user/Settings/EditProfile/AddressManagment/index.tsx

```diff
@@ -0,0 +1,55 @@
+import type { Address } from '@planet-sdk/common';
+import type { AddressAction } from './microComponents/AddressActionMenu';
+
+import { useMemo, useState } from 'react';
+import { useTranslations } from 'next-intl';
+import AddressList from './microComponents/AddressList';
+import { useUserProps } from '../../../../common/Layout/UserPropsContext';
+import WebappButton from '../../../../common/WebappButton';
+import styles from './AddressManagement.module.scss';
+
+export type AddressType = 'primary' | 'mailing' | 'other';
+
+export interface UpdatedAddress extends Address {
+  id: string;
+  type: AddressType;
+  name: string | null;
+  state: string | null;
+  isPrimary: boolean | null;
+  address2: string | null;
+}
+export const addressType = ['primary', 'mailing', 'other'];
+const AddressManagement = () => {
+  const { user } = useUserProps();
+  const tProfile = useTranslations('Profile.addressManagement');
+  const [userAddresses, setUserAddresses] = useState<UpdatedAddress[]>(
+    user?.addresses
+  ); // need to update planet-sdk to include addresses key
```

Comment:
We are not using user.addresses because it does not provide an updated list of addresses if an address is deleted from the address list. Instead, I have used the /app/addresses endpoint to fetch the user's addresses rather than relying on the addresses from the profile response. I have addressed this issue in the Address Action PR https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/2308